specification/indices/_types/IndexTemplate.ts

```diff
@@ -145,5 +145,5 @@ export class IndexTemplateSummary {
    * @availability stack since=8.19.0 stability=stable
    * @availability serverless stability=stable
    */
-  data_stream_options?: DataStreamOptionsTemplate | null
+  data_stream_options?: DataStreamOptions
```

Comment:
According to the description of `DataStreamOptionsTemplate`:
> Data stream options template contains the same information as DataStreamOptions but allows them to be set explicitly to null.

Which means it's only relevant for `PUT` requests, and this `IndexTemplateSummary`/`IndexTemplate` is only used in retrieval APIs. That's why I changed it to `DataStreamOptions`.